apps/ftu/js/late_customization.js

```diff
@@ -0,0 +1,657 @@
+            this._nextOnlineRequest.reject(new Error('offline'));
+            // re-start the wait/fetch/install process
+            if (!this._appsToInstall) {
+              this.attemptToPopulateAppList();
+            }
+          }
+          this._nextOnlineRequest = null;
+          break;
+        case 'ftu-done':
+          // abort anything pending
+          if (this._operatorInfoRequest) {
+            this._operatorInfoRequest.reject(
+                new Error('FTU complete with no viable operator'));
+            this._operatorInfoRequest = null;
+          }
+          if (this._nextOnlineRequest) {
+            this._nextOnlineRequest.reject(
+                new Error('done before connection became available'));
+            this._nextOnlineRequest = null;
+          }
+          break;
+      }
+    },
+
+    onPanelShown: function() {
+      UIManager.mainTitle.setAttribute('data-l10n-id',
+                                             'late-customization');
+      this.finalizeRender();
+      this.attemptInstalls();
+    },
+
+    handleMobileConnectionReady: function(evt) {
+      this.debug('handleMobileConnectionsReady: ', evt);
+      if (!this._operatorInfoRequest) {
+        return;
+      }
+      var info = this.getOperatorInfo();
+      if (info) {
+        this._operatorInfo = info;
+        this._operatorInfoRequest.resolve(info);
+        this._operatorInfoRequest = null;
+      }
+    },
+
+    _createElement: function(name, options) {
+      var elm = document.createElement(name);
+      if (options.id) {
+        elm.id = options.id;
+      }
+      if (options.className) {
+        elm.className = options.id;
+      }
+      if (options.dataset) {
+        for(var p in options.dataset) {
+          elm.dataset[p] = options.dataset[p];
+        }
+      }
+      if (options.attributes) {
+        for(var a in options.attributes) {
+          elm.setAttribute(a, options.attributes[a]);
+        }
+      }
+      if ('textContent' in options) {
+        elm.textContent = options.textContent;
+      }
+      return elm;
+    },
+
+    render: function() {
+      var section = this.element =
+          document.getElementById('late_customization');
+      if (!section) {
+        section = this._createElement('section', {
+          attributes: { role: 'region' },
+          id: 'late_customization'
+        });
+        this.element = section;
+      }
+
+      section.innerHTML = Sanitizer.escapeHTML `
+      <section id="late-customization-cando" class="content hidden">
+        <p id="late_customization-downloading-message"></p>
+      </section>
+      <section id="late-customization-nocando" class="content">
+        <p id="late_customization-download-later-message"></p>
+        <p id="late_customization-download-later-marketplace"></p>
+      </section>
+      <article data-type="list">
+        <ul id="late_customization-applist" class="hidden"></ul>
+      </article>`;
+
+      function byId(id) {
```

Comment:
Perhaps this could be a wrapper around mozL10n and the query selector like:

`function setTranslationById(id, label, values)`

and used below.